File: 2190-most-frequent-number-following-key-in-an-array/2190-most-frequent-number-following-key-in-an-array.py
class Solution:
    def mostFrequent(self, nums: List[int], key: int) -> int:
        mp = [0] * 1001
        for i in range(len(nums)-1):
            if nums[i] == key:
                mp[nums[i+1]]+=1
        max_cnt, res = 0,0
        for i in range(len(mp)):
            if max_cnt < mp[i]:
                max_cnt = mp[i]
                res = i
        return res
        

File: 2190-most-frequent-number-following-key-in-an-array/2190-most-frequent-number-following-key-in-an-array.cpp
class Solution {
public:
    int mostFrequent(vector<int>& nums, int key) {
        int map[1001] =  {};
        for(int i = 0; i < nums.size() - 1; i++)
        {
            if(nums[i] == key) map[nums[i+1]]++;
        }
        int max_cnt(0), res(0);
        for(int i = 0; i < 1001; i++)
        {
            if(max_cnt < map[i])
            {
                max_cnt = map[i];
                res = i;
            }
        }
        return res;
    }
};